doc/developers/porting.html

```diff
@@ -118,40 +118,101 @@ <h2>Porting to a new platform</h2>
     <h2>Cross compiling</h2>
 
     <p>
-      It is possible to use <a href="http://www.cmake.org/"
-      target="_top">CMake</a> to cross compile
-      <acronym>deal.II</acronym> for a foreign platform.
-      Further information on that topic can be found at the <a
-      href="http://www.cmake.org/Wiki/CMake_Cross_Compiling">CMake
-      wiki</a>.
+      It is possible to use the
+      <a href="https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html">CMake toolchain</a>
+      to cross compile deal.II for a platform other than the one on which the
+      compiler is running. The target platform can have a different operating
+      system, different architecture or different set of libraries. Cross
+      compilation is a very useful technique, for instance it can be used to
+      compile deal.II with a compiler that is not available in the target
+      machine. An alternative technique is to use a
+      <a href="https://www.docker.com/">Docker container</a> or a
+      <a href="https://en.wikipedia.org/wiki/Virtualization">virtual machine</a>
+      that mimics the target machine.
     </p>
 
     <p>
-      You have to set up a native deal.II build directory first and run
-      <code>make expand_instantiations_exe</code> in it. The executable is
-      needed for the build system (and obviously the cross compiled version
-      cannot be used). Locate the <code>expand_instantiations</code>
-      executable (it usually resides under
-      <code>${CMAKE_BINARY_DIR}/bin</code> and export its location with the
-      <code>PATH</code> environment variable.
+      You can use any compiler for cross compilation, although we find that
+      <a href="https://clang.llvm.org/">LLVM/clang</a> is the most versatile
+      because it is natively a cross-compiler. Below you can find an example
```

Comment:
What about slightly rephrasing to "[...], although we find that LLVM/clang is more versatile because it supports multiple architecture targets in a single executable natively"?
I would like to avoid the "natively a cross-compiler" expression. It might also be nice to link to
https://clang.llvm.org/docs/CrossCompilation.html as well.

Reply:
I see your point. We shouldn't choose any compiler. I took your phrase and replace "we find" by "might". I also rephrased other parts of the PR to keep it more neutral.